tests/test_omxplayer.py

```diff
@@ -137,10 +137,113 @@ def run_command(self, command_name, *args):
         command(*args)
 
     # Must have the prefix 'patch' for the decorators to take effect
-    def patch_and_run_omxplayer(self, Connection=Mock()):
+    def patch_and_run_omxplayer(self, Connection=Mock(), active=False):
         bus_address_finder = Mock()
         bus_address_finder.get_address.return_val = "example_bus_address"
         self.player = OMXPlayer(self.TEST_FILE_NAME,
                                 bus_address_finder=bus_address_finder,
                                 Connection=Connection,
                                 cleaner=Mock())
+        if active:
+            self.player._process.poll = Mock(return_value=None)
+
+    def test_stop_event(self, *args):
+        self.patch_and_run_omxplayer(active=True)
+                
+        def callback(player):
```

Comment:
You'd be better off just using `Mock.assert_called_once_with()` (see http://www.voidspace.org.uk/python/mock/mock.html)

That way the test verifies that the callback has only been called once and with the correct parameters rather than causing the expected side effect.

This goes for all the functions, it'll also make them much shorted and clearer